src/NUnitTestAdapter/NUnit3TestDiscoverer.cs

```diff
@@ -97,6 +91,11 @@ public void DiscoverTests(IEnumerable<string> sources, IDiscoveryContext discove
                         int cases = ProcessTestCases(topNode, discoverySink, testConverter);
 
                         TestLog.Debug(string.Format("Discovered {0} test cases", cases));
+                        // Only save if seed is not specified in runsettings
+                        // This allows workaround in case there is no valid
+                        // location in which the seed may be saved.
+                        if (cases>0 && !Settings.RandomSeedSpecified)
+                            Settings.SaveRandomSeed(Path.GetDirectoryName(sourceAssemblyPath));
```

Comment:
Have you verified that the random seed saved at this point is the same one saved initially? I think that's the case, but just want to be sure.  Also, do we have random tests?